source/computer/wsl-setup.rst

```diff
@@ -1,18 +1,202 @@
+
+简介
+----
+
+WSL 有 WSL1 和 WSL2 两个发行版本，二者底层原理不同。大多数情况下，建议使用 WSL2，
+因为它提供更快的性能和 100% 的系统调用兼容性。涉及到跨系统的文件互访时
+（Linux 访问 Windows 里的文件，或 Windows 访问 Linux 里的文件），使用 WSL1 具有
+更快的性能。
+
+官方目前没有弃用 WSL1 的计划，并且支持将任何一个已经安装的 Linux 发行版转换为 WSL1 或者 WSL2。
+
+安装 WSL
+--------
+
+WSL 的安装方式可以参考官方文档：
+
+- 中文指南：https://docs.microsoft.com/zh-cn/windows/wsl/install-win10
+- 英文指南：https://docs.microsoft.com/en-us/windows/wsl/install-win10
+
+官方文档对 WSL 安装配置中可能遇到的常见问题都给出了详细的解决方案，请认真阅读。
+
+WSL 可以安装不同的 Linux 发行版，但目前官方并未提供 Fedora 发行版的安装。
+推荐使用 WSL 安装 Ubuntu 20.04 LTS。
+
+.. note::
+
+   想在 WSL 上安装 Fedora 的读者可以参考以下安装指南：
+
+   - 英文指南：https://fedoramagazine.org/wsl-fedora-33/
+   - 中文指南：https://suiahae.me/Using-Fedora-33-on-Windows-10-WSL2/
+
+   需要注意的是，我们尚未验证以上安装指南是否有效。
+   
+   指南中的 Fedora rootfs 下载地址可能已失效。请访问 
+   https://github.com/fedora-cloud/docker-brew-fedora/tree/33/x86_64
+   下载最新版本。
+      
+
+常用命令
+--------
+
+打开 CMD 或 PowerShell，我们可以运行如下命令来使用 WSL。以下示例命令假定我们已
+安装 WSL2 版本的 Ubuntu 20.04 LTS，且其名称为 Ubuntu。
+
+``wsl`` 命令用法::
+
+    # 查看 wsl 命令帮助
+    $ wsl -h
+
+    # 列出所有已安装的 Linux 发行版的状态，并显示是 WSL1 还是 WSL2
+    $ wsl -l -v
+
+    # 停止正在运行的 Linux 发行版
+    $ wsl -t Ubuntu
+
+    # 将 Ubuntu 由 WSL2 更改为 WSL1
+    $ wsl --set-version Ubuntu 1
+
+    # 将 Ubuntu 由 WSL1 改回 WSL2
+    $ wsl --set-version Ubuntu 2
+
+    # 设置默认发行版
+    $ wsl -s Ubuntu20.04
+
+    # 删除某个发行版（如名为 Ubuntu）
+    $ wsl --unregister Ubuntu
+
+启动、进入和退出 Linux 环境::
+
+    # 启动并进入 Linux 环境（进入默认发行版）
+    $ bash
+
+    # 退出 Linux 环境（并不会改变 WSL 的运行状态）
+    $ exit
+
+开启 WSL 后，Linux 发行版的默认安装位置是 C 盘。为了避免占用 C 盘的大量空间，
+可以将已安装的 Linux 发行版导出备份，再导入还原到其他盘，最后删除 C 盘上的发行版。
+打开 PowerShell，执行如下命令::
+
+    # 导出 Linux 发行版，可做为备份
+    # 在 D 盘中新建备份目录，命名为 WSLBAK
+    $ mkdir -p D:\WSLBAK
+    # 导出到备份目录下，命名为 20210117bak.tar
+    $ wsl --export Ubuntu D:\WSLBAK\20210117bak.tar
+
+    # 导入并还原之前备份的 Linux 发行版
+    # 此例中选择在 D 盘中新建还原目录，命名为 Ubuntu20.04
+    $ mkdir -p D:\WSLDIR\Ubuntu20.04
+    # 导入并还原之前的备份，将此发行版命名为 Ubuntu20.04
+    $ wsl --import Ubuntu20.04 D:\WSLDIR\Ubuntu20.04 D:\WSLBAK\20210117bak.tar
+
+    # 删除 C 盘里名为 Ubuntu 的发行版，以释放 C 盘空间
+    $ wsl --unregister Ubuntu
+
+Windows 和 WSL 文件系统互访
+---------------------------
+
+WSL1 和 WSL2 都可以和 Windows 系统互相访问文件，但是无论从 WSL 访问 Windows，
+还是从 Windows 访问 WSL，WSL1 的速度都要远远快于 WSL2。因此，需要经常跨系统操作文件
+时，建议将 Linux 发行版设置为 WSL1。
+
+WSL 访问 Windows
+^^^^^^^^^^^^^^^^
+
+Windows 系统的硬盘挂载在 WSL 的 ``/mnt`` 路径下，用户可以在 WSL 终端中
+输入 ``cd /mnt/d`` 命令进入 Windows 系统的 D 盘，然后对目录中的文件进行编辑、运行。
+
+Windows 访问 WSL
+^^^^^^^^^^^^^^^^
+
+有两种方式可以在 Windows 中打开 WSL 的文件和目录：
+
+1. 在 Windows 资源管理器的地址栏中输入 ``\\wsl$`` 会显示所有已安装的 WSL 目录，
+   然后根据需要找到文件进行操作
+
+2. 进入 WSL 之后在终端输入 ``cd ~ && explorer.exe .`` 会在 Windows 下打开
+   家目录，根据需要找到文件进行修改
+
+由于 WSL2 使用 VHD 虚拟磁盘文件作为 Linux 发行版的根目录，导致 Windows
+的 CMD 和 PowerShell 无法直接使用真实路径访问 WSL2 文件系统。
+同时，CMD 和 PowerShell 不支持 UNC 路径（指类似 ``\\wsl$`` 这种格式的路径）。
+因此，如果想使用 Windows 的应用程序、CMD 以及 PowerShell 编译或运行 WSL 中的文件，
+需要先把 Linux 发行版切换到 WSL1 版本，进入 WSL 后新建一个名字独特的文件夹，
+然后在 Windows 中对该文件夹进行定位，从而确定其真实的路径。
```

Comment:
CMD and Powershell can visit true path on WLS1, but cannot visit true path on WSL2, right?
CMD and Powershell cannot visit UNC path on either WSL1 and WSL2, right?
How about Windows applications?

Reply:
Exactly. To the third one, Windows app can visit the true path like `d:\wsldir\ubuntu20.04\home\zhao\filename`

Reply:
You mean Windows app **CAN** visit the true path of both WSLs, but **CANNOT** visit UNC path for both WSLs

Reply:
Only WSL1 has true path.